ptvsd/wrapper.py

```diff
@@ -940,6 +941,27 @@ def on_evaluate(self, request, args):
                 variablesReference=0)
             return
 
+        if context == 'repl' and is_eval_error == 'True':
+            # try exec for repl requests
+            cmd_args = (pyd_tid, pyd_fid, 'LOCAL', expr, '1')
+            _, _, resp_args = yield self.pydevd_request(
+                pydevd_comm.CMD_EXEC_EXPRESSION,
+                msg)
+            try:
+                xml2 = untangle.parse(resp_args).xml
+                xvar2 = xml2.var
+                result_type  = unquote(xvar2['type'])
+                result = unquote(xvar2['value'])
+            except:
+                # if resp_args is not xml then it contains the error traceback
+                result_type  = unquote(xvar['type'])
```

Comment:
If we're sending the traceback, then we need to filter the traceback to hide the internals.
Currently we're sending too much (internal) information back to the user. I would have thought just the error message would suffice. what's the current behaviour in VS. In VSC we just display the error message.

Reply:
Actually the comment there is incorrect. When i had initially written this it was returning trace back. but this is what you will see:
![image](https://user-images.githubusercontent.com/3840081/37552110-b7388be2-296b-11e8-8689-7910adf2b5c0.png)

Reply:
The traceback you see in #219 comes from exec itself. We may have to do a bit of trickery to avoid this. I tried using `sys.displayhook` to capture this info but that did not help. @int19h has a few ides on how to fix that.

Reply:
I had a similar idea (when using `CMD_CONSOLE_EXEC`) but that's where I got stuck this evening. Gets complicated when you evaluate a function that starts a thread which prints stuff (sounds too far fetched). 
Anyways, would like to see how it is that the stack trace isn't printed for your code, but only for VSC. Is it because you're ignoring what comes from `output` event?

Reply:
The traceback is printed in the console output, in the REPL window we print what is returned via the protocol. the screen sot was from the REPL window.

Reply:
Interesting, will need to discuss this in further detail. (I thought the console===REPL).

Reply:
In REPL scenarios, everything that normally goes into stdout/err is supposed to go to REPL - it just isn't wired yet. But once we do, we will have that problem.

Reply:
* Out of curiosity whey aren't we using `CMD_CONSOLE_EXEC` command, after all the value for `context` is `repl` 

* >it just isn't wired yet. But once we do, we will have that problem.

    * I think I understand, VSC is getting the `stdout/stderr` messages via the protocol as VSC is passing in the value `true` for the setting [redirectOutput](https://github.com/Microsoft/ptvsd/blob/master/ptvsd/wrapper.py#L644). I'm assuming VS doesn't use this.

Reply:
I will give CMd_CONSOLE_EXEC a try in the morning, and update this PR if it seems better.

Reply:
CMD_CONSOLE_EXEC does not work the way we want it. For the following REPL statements, this is the result we get:

**In REPL window**:
```
>>> random.random()
False
>>> import random
False
>>> random.random()
False
>>> 
```

**In Console Window**:
```
Traceback (most recent call last):
  File "c:\git\ptvsd\ptvsd\pydevd\_pydevd_bundle\pydevd_exec2.py", line 3, in Exec
    exec(exp, global_vars, local_vars)
  File "<input>", line 1, in <module>
NameError: name 'random' is not defined
<xml><var name="" type="bool" qualifier="builtins" value="False" />
</xml>

<xml><var name="" type="bool" qualifier="builtins" value="False" />
</xml>

0.9210117288960727
<xml><var name="" type="bool" qualifier="builtins" value="False" />
</xml>
```


**XML response received from pydevd:**
```
>>> random.random()
<xml><var name="" type="bool" qualifier="builtins" value="False" />
</xml>

>>> import random
<xml><var name="" type="bool" qualifier="builtins" value="False" />
</xml>

>>> random.random()
<xml><var name="" type="bool" qualifier="builtins" value="False" />
</xml>
```

As you can see it always return false.

Reply:
All good at my end 👍